server/store/kvstore/upgrade_test.go

```diff
@@ -215,3 +215,166 @@ func TestUpgradeTo14(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestUpgradeTo18(t *testing.T) {
+	t.Run("Success to migrate", func(t *testing.T) {
+		poll1 := poll.Poll{
+			ID:     testutils.GetPollID(),
+			PostID: model.NewId(),
+		}
+		postWithCustomType := &model.Post{
+			Props: model.StringInterface{
+				"attachments": []*model.SlackAttachment{{
+					Actions: []*model.PostAction{
+						{Type: "custom_matterpoll_admin_button", Id: "endPoll"},
+						{Type: model.PostActionTypeButton, Id: "resetVote"},
+					},
+				}},
+			},
+		}
+		updatedPost := &model.Post{
+			Type: model.PostTypeSlackAttachment,
+			Props: model.StringInterface{
+				"attachments": []*model.SlackAttachment{{
+					Actions: []*model.PostAction{
+						{Type: model.PostActionTypeButton, Id: "endPoll"},
+						{Type: model.PostActionTypeButton, Id: "resetVote"},
+					},
+				}},
+			},
+		}
+
+		api := &plugintest.API{}
+		api.On("KVList", 0, perPage).Return([]string{pollPrefix + poll1.ID}, nil)
+		api.On("KVGet", pollPrefix+poll1.ID).Return(poll1.EncodeToByte(), nil)
+		api.On("GetPost", poll1.PostID).Return(postWithCustomType, nil)
+		api.On("UpdatePost", updatedPost).Return(updatedPost, nil)
```

Comment:
Arg, I really dislike mocked tests. They don't give me full confidence that the code works as expected.

I would love to get proper integration tests working using something like https://github.com/mattermost/mattermost-plugin-msteams/pull/563.